src/chainrules.jl

```diff
@@ -31,12 +33,15 @@ ChainRulesCore.@scalar_rule(
 )
 ChainRulesCore.@scalar_rule(dawson(x), 1 - (2 * x * Ω))
 ChainRulesCore.@scalar_rule(digamma(x), trigamma(x))
-ChainRulesCore.@scalar_rule(erf(x), (2 / sqrt(π)) * exp(-x * x))
-ChainRulesCore.@scalar_rule(erfc(x), -(2 / sqrt(π)) * exp(-x * x))
+ChainRulesCore.@scalar_rule(erf(x), (2 / sqrt(π)) * exp(-x^2))
+ChainRulesCore.@scalar_rule(erfc(x), -(2 / sqrt(π)) * exp(-x^2))
+ChainRulesCore.@scalar_rule(logerfc(x), -(2 / sqrt(π)) * exp(-x^2 - Ω))
```

Comment:
Would be good to use the new constants from #345.

Reply:
I'll update the PR now that the other one was merged.

Reply:
I updated the ChainRules definitions, they use the irrational constants as well now.

This fixes https://github.com/JuliaMath/SpecialFunctions.jl/issues/307 for most functions. However, for `erfcx` additional constants would be needed: https://github.com/JuliaMath/IrrationalConstants.jl/pull/8